apps/api/src/app/widgets/e2e/remove-messages-bulk.e2e.ts

```diff
@@ -0,0 +1,104 @@
+  let session: UserSession;
+  let template: NotificationTemplateEntity;
+  let subscriberId: string;
+  let subscriberToken: string;
+  let subscriberProfile: {
+    _id: string;
+  } | null = null;
+
+  beforeEach(async () => {
+    session = new UserSession();
+    await session.initialize();
+    subscriberId = SubscriberRepository.createObjectId();
+
+    template = await session.createTemplate({
+      noFeedId: true,
+    });
+
+    const { body } = await session.testAgent
+      .post('/v1/widgets/session/initialize')
+      .send({
+        applicationIdentifier: session.environment.identifier,
+        subscriberId,
+        firstName: 'Test',
+        lastName: 'User',
+        email: 'test@example.com',
+      })
+      .expect(201);
+
+    const { token, profile } = body.data;
+
+    subscriberToken = token;
+    subscriberProfile = profile;
+  });
+
+  it('should remove messages by bulk', async function () {
+    await session.triggerEvent(template.triggers[0].identifier, subscriberId);
+    await session.triggerEvent(template.triggers[0].identifier, subscriberId);
+    await session.triggerEvent(template.triggers[0].identifier, subscriberId);
+
+    await session.awaitRunningJobs(template._id);
+
+    const messagesBefore = await messageRepository.find({
+      _environmentId: session.environment._id,
+      _subscriberId: subscriberProfile?._id,
+      channel: ChannelTypeEnum.IN_APP,
+    });
+
+    expect(messagesBefore.length).to.equal(3);
+
+    const [firstMessage, ...messagesToDelete] = messagesBefore;
+
+    await axios.post(
+      `http://127.0.0.1:${process.env.PORT}/v1/widgets/messages/bulk/delete`,
+      { messageIds: messagesToDelete.map((msg) => msg._id) },
+      {
+        headers: {
+          Authorization: `Bearer ${subscriberToken}`,
+        },
+      }
+    );
+
+    const messagesAfter = await messageRepository.find({
+      _environmentId: session.environment._id,
+      _subscriberId: subscriberProfile?._id,
+      channel: ChannelTypeEnum.IN_APP,
+    });
+
+    expect(messagesAfter.length).to.equal(1);
+    expect(messagesAfter[0]._id).to.equal(firstMessage._id);
+  });
+
+  it('should throw exception id not messages were provided', async function () {
+    await session.triggerEvent(template.triggers[0].identifier, subscriberId);
+    await session.triggerEvent(template.triggers[0].identifier, subscriberId);
+    await session.triggerEvent(template.triggers[0].identifier, subscriberId);
+
+    await session.awaitRunningJobs(template._id);
+
+    try {
+      const res = await axios.post(
+        `http://127.0.0.1:${process.env.PORT}/v1/widgets/messages/bulk/delete`,
+        {},
+        {
+          headers: {
+            Authorization: `Bearer ${subscriberToken}`,
+          },
+        }
+      );
+
+      expect(true).to.equal(false);
+    } catch (e) {
+      expect(e.response.data.message).to.contain('messageIds should not be empty');
+    }
+  });
```

Comment:
I thought about it, but was not sure if it should be on the API level, but thinking at it again it would be a good test to test the interface in case someone changes the decorator/dto by mistake.